lib/frame_composer/ffmpeg_research_compositor_module.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Membrane.VideoCompositor.FFMPEG.Research do
+  @moduledoc """
+  This module implements video composition in ffmpeg.
+  """
+  @behaviour Membrane.VideoCompositor.FrameCompositor
+
+  alias Membrane.VideoCompositor.FFmpeg.Native, as: FFmpeg
+  alias Membrane.VideoCompositor.FFmpeg.Native.RawVideo, as: NativeRawVideo
+
+  @impl Membrane.VideoCompositor.FrameCompositor
+  def init(caps) do
+    {:ok, video} = NativeRawVideo.from_membrane_raw_video(caps)
+
+    videos = [video, video]
+
+    {:ok, internal_state} = FFmpeg.init(videos)
+    {:ok, %{state: internal_state, iter: 1, raw_video: video}}
+  end
+
+  @impl Membrane.VideoCompositor.FrameCompositor
+  def merge_frames(frames, state_of_init_module) do
+    %{state: internal_state, iter: iter} = state_of_init_module
+
+    repeated_frames = Stream.repeatedly(fn -> frames.second end) |> Enum.take(div(iter, 80))
+    frames = [frames.first, frames.second] ++ repeated_frames
+
+    internal_state =
+      if rem(iter, 80) == 0 do
+        raw_videos = for _n <- 1..(div(iter, 80) + 2), do: state_of_init_module.raw_video
+
+        {:ok, new_internal_state} = FFmpeg.init(raw_videos)
```

Comment:
NOT: would be nic if you would quickly describe it in the moduledoc - or is it temporary and you are going to change the algorithm in the future?